### PR TITLE
Accelerate strategy

### PR DESCRIPTION
**Pull request by kbatoev**

### CODE REVIEW

File: accelerateStrategy.cpp [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+	};
+
+	cntPowers = {
+		{X1, 0}
+		, {Y1, 0}
+		, {X2, 0}
+		, {Y2, 0}
+	};
+
+	indeces = {
```
Comment (iakov) [inline accelerateStrategy.cpp]: indices?

File: accelerateStrategy.cpp [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+	, pad2WasActive(false)
+	, speed(300)
+	, padsMapper(new QSignalMapper(this))
+{
+
+	connect(&stopTimerForPad1, SIGNAL(timeout()), padsMapper, SLOT(map()));
+	padsMapper->setMapping(&stopTimerForPad1, 1);
+	connect(&stopTimerForPad1, SIGNAL(timeout()), &stopTimerForPad1, SLOT(stop()));
+
+	connect(&stopTimerForPad2, SIGNAL(timeout()), padsMapper, SLOT(map()));
```
Comment (iakov) [inline accelerateStrategy.cpp:33]: Might be single timer for both pads is enough?

File: accelerateStrategy.cpp [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+	}
+
+	if (!pad1WasActive && !pad2WasActive)
+		workTimer.stop();
+}
+
+void AccelerateStrategy::dealWithPads()
+{
+
+	if (mPressedKeys.size()) {
```
Comment (iakov) [inline accelerateStrategy.cpp]: Must be !mPressedKeys.empty(). Never use .size() to test if collection is empty

**Review by iakov — CHANGES_REQUESTED** [review]

File: accelerateStrategy.cpp (hunk 2 above)
Comment (kbatoev) [reply accelerateStrategy.cpp:33]: There was one pad for both pads in the previous version. The problem was when I pressed pad1, then pressed pad2, released pad1, and released pad2 in some time (ex. 30 seconds). And only in the end there were two commands: pad 1 up, pad 2 up, but I expected command "pad 1 up", when I released pad1.
So, I decided to use timer per pad.

**Pull request merged**